src/models/v3/operation-trait.ts

```diff
@@ -0,0 +1,100 @@
+import { BaseModel } from '../base';
+// import { Channels } from '../channels';
+// import { Channel } from './channel';
+import { SecurityScheme } from './security-scheme';
+import { SecurityRequirements } from '../security-requirements';
+import { SecurityRequirement } from './security-requirement';
+
+import { bindings, hasDescription, description, extensions, hasExternalDocs, externalDocs, tags } from './mixins';
+
+import type { BindingsInterface } from '../bindings';
+import type { ExtensionsInterface } from '../extensions';
+import type { ExternalDocumentationInterface } from '../external-docs';
+import type { ChannelsInterface } from '../channels';
+import type { OperationAction } from '../operation';
+import type { OperationTraitInterface } from '../operation-trait';
+import type { TagsInterface } from '../tags';
+
+import type { v3 } from '../../spec-types';
+
+export class OperationTrait<J extends v3.OperationTraitObject = v3.OperationTraitObject> extends BaseModel<J, { id: string | undefined }> implements OperationTraitInterface {
+  id(): string | undefined {
+    return this._meta.id;
+  }
+
+  hasId(): boolean {
+    return this.id() !== undefined;
+  }
+
+  action(): OperationAction | undefined {
+    return this._json.action;
+  }
+
+  hasSummary(): boolean {
+    return !!this._json.summary;
+  }
+
+  summary(): string | undefined {
+    return this._json.summary;
+  }
+
+  hasDescription(): boolean {
+    return hasDescription(this);
+  }
+
+  description(): string | undefined {
+    return description(this);
+  }
+
+  hasExternalDocs(): boolean {
+    return hasExternalDocs(this);
+  }
+
+  externalDocs(): ExternalDocumentationInterface | undefined {
+    return externalDocs(this);
+  }
+
+  isSend(): boolean {
+    return this.action() === 'send';
+  }
+
+  isReceive(): boolean {
+    return this.action() === 'receive';
+  }
+
+  channels(): ChannelsInterface {
```

Comment:
Not that function returns empty collection, because we should discuss https://github.com/asyncapi/spec/issues/856 problem. Also in current implementation that function creates circular references between import:

```
OperationTrait -> Channel -> Operation -> OperationTrait
```

I will try to fix it if we will agree what about `channel` and `action` in Operation Trait Object.